tests/e2e/specs/upgrade/upload-plugin-new-version.test.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import path from 'path';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	visitAdminPage,
+	installPlugin,
+	uninstallPlugin,
+	deactivatePlugin,
+	__experimentalRest as rest,
+} from "@wordpress/e2e-test-utils";
+import { addQueryArgs } from "@wordpress/url";
+
+async function getPlugins() {
+	const response = await rest({
+		method: 'GET',
+		path: "/wp/v2/plugins",
+	});
+	return response;
+}
+
+async function checkPluginStatus(pluginName) {
+	const plugins = await getPlugins();
+	const plugin = plugins.find(plugin => plugin.name === pluginName);
+
+	return plugin ? plugin.status : 'uninstalled';
+}
+
+async function checkAndInstallPlugin(pluginSlug, pluginName) {
+	const pluginStatus = await checkPluginStatus(pluginName);
+
+	if (pluginStatus === 'active') {
+		await deactivatePlugin(pluginSlug);
+		await uninstallPlugin(pluginSlug);
+		await installPlugin(pluginSlug, pluginName);
+	} else if (pluginStatus === 'inactive') {
+		await uninstallPlugin(pluginSlug);
+		await installPlugin(pluginSlug, pluginName);
+	} else {
+		await installPlugin(pluginSlug, pluginName);
+	}
+}
+
+async function uploadNewPluginVersion(pluginPath) {
+	const uploadPluginUrl = addQueryArgs('', { tab: 'upload' });
+	await visitAdminPage('plugin-install.php', uploadPluginUrl);
+	const input = await page.$('#pluginzip');
+	await input.uploadFile(pluginPath);
+	await page.click('#install-plugin-submit');
+}
+
+describe('Manage uploading new plugin/theme version', () => {
+	const pluginSlug = 'classic-editor';
+	const pluginName = 'Classic Editor';
```

Comment:
Could we use a smaller and test-purposed plugin instead? It will have some benefits:

- We'll be able to manage the zip source and check it into the source control as well.
- The plugin name can be explicitly named for testing purposes.
- We won't mess with the download count in wp.org.
- It will be clear in the test whether the installation is an upgrade or a downgrade.